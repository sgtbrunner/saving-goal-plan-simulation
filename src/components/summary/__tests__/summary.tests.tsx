import { screen, RenderResult } from '@testing-library/react';
import {
  MONTHLY_AMOUNT,
  SUMMARY_TEXT,
  renderWithRedux,
  getDefaultSavingGoal,
} from '../../../utils/test.utils';
import Summary, { formatUSDCurrency } from '../summary.component';

const mountComponent = (): RenderResult => renderWithRedux(<Summary />);

describe('Summary component', () => {
  const savingGoal = getDefaultSavingGoal();

  it('should match snapshot', () => {
    const { container } = mountComponent();

    expect(container).toMatchSnapshot();
  });

  it('should render the default monthly amount', () => {
    mountComponent();

    const monthlyAmount = screen.getByTestId(MONTHLY_AMOUNT);
    expect(monthlyAmount).toHaveTextContent(/monthly amount/i);
    expect(monthlyAmount).toHaveTextContent(
      formatUSDCurrency({ value: savingGoal.monthlyAmount })
    );
  });

  it('should render the correct summary text', () => {
    mountComponent();
    const { monthlyDeposits, goalAmount, reachDateMonth, reachDateYear } =
      savingGoal;

    const summaryText = screen.getByTestId(SUMMARY_TEXT);
    expect(summaryText).toHaveTextContent(
      `Youâ€™re planning ${monthlyDeposits} monthly deposits to reach your ${formatUSDCurrency(
        { value: goalAmount }
      )} goal by ${reachDateMonth} ${reachDateYear}.`
    );
  });
});
