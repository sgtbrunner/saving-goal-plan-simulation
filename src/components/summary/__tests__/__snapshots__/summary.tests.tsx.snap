// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary component should match snapshot 1`] = `
<div>
  <div
    class="sc-dlnjwi gsoFCl"
  >
    <div
      class="sc-hKFxyN iCJaxk"
      data-testid="monthly-amount"
    >
      <h4>
        Monthly amount
      </h4>
      <p>
        $25,000
      </p>
    </div>
    <div
      class="sc-eCApnc hVfgZM"
      data-testid="summary-text"
    >
      <p>
        Youâ€™re planning 
        <strong>
          1 monthly deposits
        </strong>
         
        to reach your
         
        <strong>
          $25,000
        </strong>
         goal by
         
        <strong>
          December
           
          2021
        </strong>
        .
      </p>
    </div>
  </div>
</div>
`;
